iceoryx2-bb/cxx/CMakeLists.txt

```diff
@@ -0,0 +1,40 @@
+# Copyright (c) 2024 Contributors to the Eclipse Foundation
+#
+# See the NOTICE file(s) distributed with this work for additional
+# information regarding copyright ownership.
+#
+# This program and the accompanying materials are made available under the
+# terms of the Apache Software License 2.0 which is available at
+# https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+# which is available at https://opensource.org/licenses/MIT.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+
+cmake_minimum_required(VERSION 3.22)
+
+project(iceoryx2-bb VERSION ${IOX2_VERSION_STRING} LANGUAGES CXX)
+
+set(PREFIX iceoryx2/v${CMAKE_PROJECT_VERSION})
+
+# include only lib -> includes are installed only once despite being used in static lib as well as shared lib
+
+add_library(iceoryx2-bb-containers INTERFACE)
+add_library(iceoryx2-cxx::iceoryx2-bb-containers ALIAS iceoryx2-bb-containers)
```

Comment:
The namespace should probably be the same like the project name, else it will be confusing since the user would need to do a `find_package(iceoryx2-bb)` but need to link towards `iceoryx2-cxx::*`. If we stick with just `iceoryx2-bb` as package name, I would suggest to use `iceoryx2-bb::container`.

Reply:
... well, calling it just `container` would require that the library is also called just `containers`. This should not be a big issue since it it a header only lib and if it becomes a binary lib at some point, we can always use `OUTPUT_NAME` to rename the binary to `iceoryx2_bb_containers.so`

Reply:
Completely agree but please use the name `iceoryx2-bb-container-cxx`

Reply:
You would call the namespace `iceoryx2-bb-container-cxx`?

I would suggest to not got this route and just use `iceoryx2-bb` as namespace and `container-cxx` as package name. The user would then just have to do a `find_package(iceoryx2-bb) ` and link towards `iceoryx2-bb::container-cxx`.

This means there would also be just one cmake package for `iceoryx2-bb` and not multiple ones. There would be just additional libraries in the `iceoryx2-bb` package.

Reply:
I settled for `iceoryx2-bb-cxx::iceoryx2-bb-containers-cxx` for now. It's somewhat verbose, but it does seem sensible to me?